From bc8268bf8799e23e6d4a8ba4beeea599968cd258 Mon Sep 17 00:00:00 2001
From: Todor Trundev <ttrundev@antelopeaudio.com>
Date: Tue, 21 Jul 2026 10:22:44 +0300
Subject: [PATCH 1/1] Terrain ridge style using [size] tag

Use an explicitly enumerated [size=<l>] property to separate ridges
[natural=ridge] by extent.
---
 data/mapcss-mapping.csv                       |  9 +++++
 data/styles/default/dark/colors.mapcss        |  6 +++
 data/styles/default/include/Basemap.mapcss    | 39 +++++++++++++++++++
 .../default/include/priorities_3_FG.prio.txt  |  9 +++++
 .../include/priorities_4_overlays.prio.txt    |  9 +++++
 data/styles/default/light/colors.mapcss       |  6 +++
 .../outdoors/include/priorities_3_FG.prio.txt |  9 +++++
 .../include/priorities_4_overlays.prio.txt    |  9 +++++
 8 files changed, 96 insertions(+)

diff --git a/data/mapcss-mapping.csv b/data/mapcss-mapping.csv
index d66a0e07..d1b1a65e 100644
--- a/data/mapcss-mapping.csv
+++ b/data/mapcss-mapping.csv
@@ -1773,3 +1773,12 @@ natural|volcano|active;[natural=volcano][volcano:status=active];;;;1650;
 building|construction;1651;
 roller_coaster|track;1652;
 man_made|ceremonial_gate;1653;
+natural|ridge|size|6;[natural=ridge][size=6],[natural=ridge][size=5],[natural=ridge][size=4],[natural=ridge][size=3],[natural=ridge][size=2],[natural=ridge][size=1];;name;int_name;1654;
+natural|ridge|size|7;[natural=ridge][size=7];;name;int_name;1655;
+natural|ridge|size|8;[natural=ridge][size=8];;name;int_name;1656;
+natural|ridge|size|9;[natural=ridge][size=9];;name;int_name;1657;
+natural|ridge|size|10;[natural=ridge][size=10];;name;int_name;1658;
+natural|ridge|size|11;[natural=ridge][size=11];;name;int_name;1659;
+natural|ridge|size|12;[natural=ridge][size=12];;name;int_name;1660;
+natural|ridge|size|13;[natural=ridge][size=13];;name;int_name;1661;
+natural|ridge|size|14;[natural=ridge][size=14];;name;int_name;1662;
diff --git a/data/styles/default/dark/colors.mapcss b/data/styles/default/dark/colors.mapcss
index 0a9dda35..f6db643e 100644
--- a/data/styles/default/dark/colors.mapcss
+++ b/data/styles/default/dark/colors.mapcss
@@ -208,3 +208,9 @@
 @label_isoline100: #496149;
 @label_isoline50: #496149;
 @label_isoline10: #496149;
+
+/* 7.x.TERRAIN RIDGES COLORS */
+
+@ridge: #C20000;
+@ridge_casing: darkgrey;
+@label_ridge: #720808;
diff --git a/data/styles/default/include/Basemap.mapcss b/data/styles/default/include/Basemap.mapcss
index fa95a6c0..e2b20270 100644
--- a/data/styles/default/include/Basemap.mapcss
+++ b/data/styles/default/include/Basemap.mapcss
@@ -776,3 +776,42 @@ line|z18-[isoline=step_50],
 line|z18-[isoline=zero],
 line|z18-[isoline=step_10],
 {width: 0.8; opacity: 0.4;}
+
+/* 9.x TERRAIN RIDGES */
+/* width is <z> - <size> */
+
+line|z10[natural=ridge][size=10],
+line|z11[natural=ridge][size=11],
+line|z12[natural=ridge][size=12],
+line|z13[natural=ridge][size=13],
+line|z14[natural=ridge][size=14],
+{color: @ridge;width: .5;opacity: 0.8;dashes: 10,5;}
+line|z10[natural=ridge][size=9],
+line|z11[natural=ridge][size=10],
+line|z12[natural=ridge][size=11],
+line|z13[natural=ridge][size=12],
+line|z14[natural=ridge][size=13],
+line|z15-[natural=ridge][size=14],
+{color: @ridge;width: 1;opacity: 0.8;}
+line|z9[natural=ridge][size=7],
+line|z10[natural=ridge][size=8],
+line|z11[natural=ridge][size=9],
+line|z12[natural=ridge][size=10],
+line|z13[natural=ridge][size=11],
+line|z14[natural=ridge][size=12],
+line|z15-[natural=ridge][size=13],
+{
+  color: @ridge;width: 2;opacity: 0.8;
+  text:name;font-size: 9;fill-opacity: 0.7;text-color: @label_ridge;text-position: line;
+}
+line|z9-[natural=ridge][size<=6],
+line|z10-[natural=ridge][size=7],
+line|z11-[natural=ridge][size=8],
+line|z12-[natural=ridge][size=9],
+line|z13-[natural=ridge][size=10],
+line|z14-[natural=ridge][size=11],
+line|z15-[natural=ridge][size=12],
+{
+  color: @ridge;width: 3;opacity: 0.8;casing-width: 1;casing-color: @ridge_casing;
+  text:name;font-size: 9;fill-opacity: 0.7;text-color: @label_ridge;text-position: line;
+}
diff --git a/data/styles/default/include/priorities_3_FG.prio.txt b/data/styles/default/include/priorities_3_FG.prio.txt
index 792edadd..6ecd2fd7 100644
--- a/data/styles/default/include/priorities_3_FG.prio.txt
+++ b/data/styles/default/include/priorities_3_FG.prio.txt
@@ -460,6 +460,15 @@ isoline-step_1000                                   # line z11- (also has pathte
 isoline-step_50                                     # line z14- (also has pathtext z16-)
 isoline-step_500                                    # line z11- (also has pathtext z12-)
 isoline-zero                                        # line z15- (also has pathtext z17-)
+natural-ridge-size-10                               # line z10- (also has line(casing) z13-, pathtext z12-)
+natural-ridge-size-11                               # line z11- (also has line(casing) z14-, pathtext z13-)
+natural-ridge-size-12                               # line z12- (also has line(casing) z15-, pathtext z14-)
+natural-ridge-size-13                               # line z13- (also has pathtext z15-)
+natural-ridge-size-14                               # line z14-
+natural-ridge-size-6                                # line z9- (also has line(casing) z9-, pathtext z9-)
+natural-ridge-size-7                                # line z9- (also has line(casing) z10-, pathtext z9-)
+natural-ridge-size-8                                # line z10- (also has line(casing) z11-, pathtext z10-)
+natural-ridge-size-9                                # line z10- (also has line(casing) z12-, pathtext z11-)
 === 60
 
 natural-wetland                                     # area z11- (also has caption z16-)
diff --git a/data/styles/default/include/priorities_4_overlays.prio.txt b/data/styles/default/include/priorities_4_overlays.prio.txt
index af9e7b97..6da31ed4 100644
--- a/data/styles/default/include/priorities_4_overlays.prio.txt
+++ b/data/styles/default/include/priorities_4_overlays.prio.txt
@@ -1120,6 +1120,15 @@ natural-spring-drinking_water_no                    # icon z14- (also has captio
 
 isoline-step_10                                     # pathtext z17- (also has line z15-)
 isoline-zero                                        # pathtext z17- (also has line z15-)
+natural-ridge-size-10                               # pathtext z12- (also has line z10-, line(casing) z13-)
+natural-ridge-size-11                               # pathtext z13- (also has line z11-, line(casing) z14-)
+natural-ridge-size-12                               # pathtext z14- (also has line z12-, line(casing) z15-)
+natural-ridge-size-13                               # pathtext z15- (also has line z13-)
+natural-ridge-size-14                               # WARNING: no drule defined for the priority (also has line z14-)
+natural-ridge-size-6                                # pathtext z9- (also has line(casing) z9-, line z9-)
+natural-ridge-size-7                                # pathtext z9- (also has line z9-, line(casing) z10-)
+natural-ridge-size-8                                # pathtext z10- (also has line z10-, line(casing) z11-)
+natural-ridge-size-9                                # pathtext z11- (also has line z10-, line(casing) z12-)
 === 1000
 
 amenity-grave_yard                                  # icon z17- (also has caption(optional) z17-, area z14-)
diff --git a/data/styles/default/light/colors.mapcss b/data/styles/default/light/colors.mapcss
index 707ababf..bb2856b3 100644
--- a/data/styles/default/light/colors.mapcss
+++ b/data/styles/default/light/colors.mapcss
@@ -208,3 +208,9 @@
 @label_isoline100: #496149;
 @label_isoline50: #567556;
 @label_isoline10: #567556;
+
+/* 7.x.TERRAIN RIDGES COLORS */
+
+@ridge: #C20000;
+@ridge_casing: lightgrey;
+@label_ridge: #720808;
diff --git a/data/styles/outdoors/include/priorities_3_FG.prio.txt b/data/styles/outdoors/include/priorities_3_FG.prio.txt
index 8dc1dce0..6b0b03a8 100644
--- a/data/styles/outdoors/include/priorities_3_FG.prio.txt
+++ b/data/styles/outdoors/include/priorities_3_FG.prio.txt
@@ -462,6 +462,15 @@ isoline-step_1000                                   # line z10- (also has pathte
 isoline-step_50                                     # line z13- (also has pathtext z15-)
 isoline-step_500                                    # line z10- (also has pathtext z11-)
 isoline-zero                                        # line z15- (also has pathtext z15-)
+natural-ridge-size-10                               # line z10- (also has line(casing) z13-, pathtext z12-)
+natural-ridge-size-11                               # line z11- (also has line(casing) z14-, pathtext z13-)
+natural-ridge-size-12                               # line z12- (also has line(casing) z15-, pathtext z14-)
+natural-ridge-size-13                               # line z13- (also has pathtext z15-)
+natural-ridge-size-14                               # line z14-
+natural-ridge-size-6                                # line z9- (also has line(casing) z9-, pathtext z9-)
+natural-ridge-size-7                                # line z9- (also has line(casing) z10-, pathtext z9-)
+natural-ridge-size-8                                # line z10- (also has line(casing) z11-, pathtext z10-)
+natural-ridge-size-9                                # line z10- (also has line(casing) z12-, pathtext z11-)
 === 60
 
 natural-wetland                                     # area z11- (also has caption z16-)
diff --git a/data/styles/outdoors/include/priorities_4_overlays.prio.txt b/data/styles/outdoors/include/priorities_4_overlays.prio.txt
index cb6ba0e2..4b2ca751 100644
--- a/data/styles/outdoors/include/priorities_4_overlays.prio.txt
+++ b/data/styles/outdoors/include/priorities_4_overlays.prio.txt
@@ -1122,6 +1122,15 @@ natural-spring-drinking_water_no                    # icon z12- (also has captio
 
 isoline-step_10                                     # pathtext z16- (also has line z15-)
 isoline-zero                                        # pathtext z15- (also has line z15-)
+natural-ridge-size-10                               # pathtext z12- (also has line z10-, line(casing) z13-)
+natural-ridge-size-11                               # pathtext z13- (also has line z11-, line(casing) z14-)
+natural-ridge-size-12                               # pathtext z14- (also has line z12-, line(casing) z15-)
+natural-ridge-size-13                               # pathtext z15- (also has line z13-)
+natural-ridge-size-14                               # WARNING: no drule defined for the priority (also has line z14-)
+natural-ridge-size-6                                # pathtext z9- (also has line(casing) z9-, line z9-)
+natural-ridge-size-7                                # pathtext z9- (also has line z9-, line(casing) z10-)
+natural-ridge-size-8                                # pathtext z10- (also has line z10-, line(casing) z11-)
+natural-ridge-size-9                                # pathtext z11- (also has line z10-, line(casing) z12-)
 power-line                                          # pathtext z15- (also has line z13-, line::dash z13-)
 === 1000
 
-- 
2.54.0.windows.1

